platform/ios/src/MGLMapView.mm

```diff
@@ -3824,10 +3824,14 @@ - (void)setShowsUserLocation:(BOOL)showsUserLocation
         if ([self.delegate respondsToSelector:@selector(mapView:viewForAnnotation:)])
         {
             userLocationAnnotationView = (MGLUserLocationAnnotationView *)[self.delegate mapView:self viewForAnnotation:self.userLocation];
-            if (userLocationAnnotationView)
+            if (userLocationAnnotationView && ! [userLocationAnnotationView isKindOfClass:MGLUserLocationAnnotationView.class])
             {
-                NSAssert([userLocationAnnotationView.class isSubclassOfClass:MGLUserLocationAnnotationView.class],
-                         @"User location annotation view must be a subclass of MGLUserLocationAnnotationView");
+                // Ignore the user-provided annotation if it is not a kind of
+                // MGLUserLocationAnnotationView (and use the default user
+                // location annotation instead).
+                userLocationAnnotationView = nil;
+
+                NSLog(@"Ignoring incorrectly-typed user location annotation view. User location annotation view must be a kind of MGLUserLocationAnnotationView. This warning will become an error in a future version.");
```

Comment:
https://github.com/mapbox/mapbox-gl-native/pull/6163/commits/af771e0294543d707b596ad84ae126e000170583 limits the warning to being shown once and adds the name of the incorrect view to the text.